src/netaddress.cpp

```diff
@@ -221,7 +222,7 @@ bool CNetAddr::IsValid() const
 
 bool CNetAddr::IsRoutable() const
 {
-    return IsValid() && !(IsRFC1918() || IsRFC2544() || IsRFC3927() || IsRFC4862() || IsRFC6598() || IsRFC5737() || (IsRFC4193() && !IsTor()) || IsRFC4843() || IsLocal());
+    return IsValid() && (Params().NetworkIDString() == CBaseChainParams::REGTEST || !(IsRFC1918() || IsRFC2544() || IsRFC3927() || IsRFC4862() || IsRFC6598() || IsRFC5737() || (IsRFC4193() && !IsTor()) || IsRFC4843() || IsLocal()));
```

Comment:
I don't think it's a good idea. We are trying to (slowly) get rid of the logic like that actually. I would rather see it somewhere where `IsRoutable()` is avoided specifically, not canceling function logic from the inside. One (preferable) way to do so would be to add smth like protected `fRequireRoutable` to `CChainParams`, specifying different values for mainnet/testnet/regtest, wrapping access into public function and using it via `Params()`. See `fMiningRequiresPeers` as as example. But again, this class is a base one and should be kept network agnostic, so rather use `Params()` where you want `IsRoutable()` to be avoided specifically.

Reply:
Agree. Actually we can explicitly specify `-externalip` in tests and then `IsRoutable` should be skipped only at initialization phase: https://github.com/dashpay/dash/pull/1608/files#diff-9a82240fe7dfe86564178691cc57f2f1L207
Is it acceptable solution, what do you think?

Reply:
Hmmm... let's take a step back actually - what issue exactly are you trying to solve by that?

Reply:
The issue is that masternodes need to be accessible from the Internet, and local network addresses (like 127..., 192.168..., 172... etc) aren't considered as valid external addresses. So we need to make an exception and allow these addresses for our automatic tests.

Reply:
Than maybe it's `CMasternode::IsValidNetAddr` that have to be fixed/adjusted instead?

Reply:
It seems this method is already adjusted and always returns `false` for regtest. The issue with masternodes is here: https://github.com/dashpay/dash/blob/91ae0b712afca7c27f0e3729b6b5a5a7810970de/src/activemasternode.cpp#L175

Reply:
BTW, this line also seems like workaround for explicitly specified addresses: https://github.com/the-schnibble/dash/blob/fe491f97868ff36a31646cbee7ad09a284b8beee/src/net.cpp#L210